dll/win32/shell32/CShellLink.cpp

```diff
@@ -2752,6 +2752,7 @@ BOOL CShellLink::OnInitDialog(HWND hwndDlg, HWND hwndFocus, LPARAM lParam)
           m_sIcoPath, m_sPath, m_sPathRel, sProduct, m_sWorkDir);
 
     m_bInInit = TRUE;
+    UINT darwin = m_Header.dwFlags & (SLDF_HAS_DARWINID);
```

Comment:
Why do you initialize this that far away from where it's going to be used (line 2841)?
As far as I can see, the `dwFlags` member doesn't seem to be modified down below.
If so, you could just have a `BOOL disablecontrols = (m_Header.dwFlags & (SLDF_HAS_DARWINID));` down below.

Reply:
Because Darwin is going to need special initialization of the text fields up here but I did not have access to a Darwin link for testing. If someone knows an app in Rapps that creates such a shortcut, that would be great...

Reply:
If you create/have a jira ticket for it,
add a DPRINT1/ ERR logging when a darwin link is created to report said application in the jira ticket.